src/fields/field.impl.hh

```diff
@@ -682,6 +685,34 @@ std::shared_ptr< FieldFE<spacedim, Value> > Field<spacedim,Value>::get_field_fe(
 }
 
 
+template<int spacedim, class Value>
+void Field<spacedim, Value>::cache_allocate(EvalSubset sub_set) {
+    unsigned int point_dim = sub_set.eval_points().point_dim();
+
+    value_cache_[point_dim-1] = FieldValueCache<Value>(sub_set.eval_points());
+    value_cache_[point_dim-1].mark_used(sub_set);
+}
+
+
+template<int spacedim, class Value>
+void Field<spacedim, Value>::cache_update(ElementCacheMap &cache_map, EvalPoints &eval_points) {
+    unsigned int dim = cache_map.dim();
+    FieldValueCache<Value> &dim_cache = value_cache_[dim-1];
+    const std::vector<unsigned int> &added_elements = cache_map.added_elements();
+    const std::set<int> &used_points = dim_cache.used_points();
+
+    for (unsigned int elm_idx : added_elements) {
```

Comment:
We need to use DHCellAccessors instead of ElementAccessors in field evaluation. DHCellAccessor contains optional element cache index (see previous comment) that allows to get position in FieldValueCache.

We can solve it at the next meeting.